packages/filebrowser/src/listing.ts

```diff
@@ -860,6 +882,22 @@ class DirListing extends Widget {
     default:
       break;
     }
+
+    // Detects printable characters typed by the user.
+    // Not all browsers support .key, but it discharges us from reconstructing
+    // characters from key codes.
+    if (event.key !== undefined && event.key.length === 1) {
```

Comment:
I'm not sure this would probably detect, e.g. `_`? from https://github.com/phosphorjs/phosphor/blob/297d7d247d69de6d329336ab6e30fd26bd4c204f/packages/keyboard/src/index.ts#L143 it looks like it's only switching on `.keyCode`, while it would also need to look at which modifiers are pressed (since `_` is <kbd>Shift</kbd>+<kbd>-</kbd>).